staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/grpc_service_unix_test.go

```diff
@@ -30,12 +30,39 @@ import (
 	"google.golang.org/grpc"
 
 	kmsapi "k8s.io/apiserver/pkg/storage/value/encrypt/envelope/v1beta1"
+	"sync"
+	"time"
 )
 
 const (
 	endpoint = "unix:///@kms-socket.sock"
 )
 
+// TestKMSPluginLateStart tests the scenario where kms-plugin pod/container starts after kube-apiserver pod/container.
+func TestKMSPluginLateStart(t *testing.T) {
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		wg.Done()
+		service, err := NewGRPCService(endpoint)
+		if err != nil {
+			t.Fatalf("failed to create envelope service, error: %v", err)
+		}
+		defer destroyService(service)
+	}()
+
+	// We know that goroutine that creates the service is running.
+	wg.Wait()
```

Comment:
This should be in a defer or at the bottom of test. Otherwise you wait for the above goroutine to finish before proceeding.